kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/checksum/MetadataChecksumGenerator.java

```diff
@@ -30,7 +34,7 @@ default void appendMetadata(HasMetadata entity) {
         appendMetadata(entity.getMetadata());
     }
 
-    void appendString(String value);
+    void appendString(@Nullable String value);
```

Comment:
It's annoying that Java won't let you use `protected` on interface methods, because the API exposed by this interface is actually much smaller than it appears. The only workaround would be to turn this `interface` into an `abstract class`.

Reply:
Yeah. I'm not entirely sure what I think the better option is. I default to interfaces rather than abstract classes so I'm inclined to leave it as it is for now. Might be worth revisiting if we ever want to support a SHA1 or other checksum.